util/staticfs/static.go

```diff
@@ -44,11 +45,11 @@ func (fs *FS) Add(p string, stat types.Stat, data []byte) {
 func (fs *FS) Walk(ctx context.Context, fn filepath.WalkFunc) error {
 	keys := make([]string, 0, len(fs.files))
 	for k := range fs.files {
-		keys = append(keys, k)
+		keys = append(keys, convertPathToKey(k))
 	}
 	sort.Strings(keys)
 	for _, p := range keys {
-		st := fs.files[p].Stat
+		st := fs.files[convertKeyToPath(p)].Stat
 		if err := fn(p, &fsutil.StatInfo{Stat: &st}, nil); err != nil {
```

Comment:
first parameter to `fn` should also be with `/`iiuc. 

Better to rename `s/p/k/` here as it is not the path anymore.